modules/core/src/main/java/org/locationtech/jts/geom/CoordinateSequence.java

```diff
@@ -45,22 +45,76 @@
 public interface CoordinateSequence
     extends Cloneable
 {
-  /**
-   * Standard ordinate index values
-   */
+  /** Standard ordinate index value for, where X is 0 */
   int X = 0;
+
+  /** Standard ordinate index value for, where Y is 1 */
   int Y = 1;
+  
+  /**
+   * Standard ordinate index value for, where Z is 2.
+   *
+   * <p>This constant assumes XYZM coordinate sequence definition, please check this assumption
+   * using {@link #getDimension()} and {@link #getMeasures()} before use.
+   */
   int Z = 2;
+
+  /**
+   * Standard ordinate index value for, where M is 3.
+   *
+   * <p>This constant assumes XYZM coordinate sequence definition, please check this assumption
+   * using {@link #getDimension()} and {@link #getMeasures()} before use.
+   */
   int M = 3;
 
   /**
-   * Returns the dimension (number of ordinates in each coordinate)
-   * for this sequence.
+   * Returns the dimension (number of ordinates in each coordinate) for this sequence.
+   *
+   * <p>This total includes any measures, indicated by non-zero {@link #getMeasures()}.
    *
    * @return the dimension of the sequence.
    */
   int getDimension();
 
+  /**
+   * Returns the number of measures included in {@link #getDimension()} for each coordinate for this
+   * sequence.
+   * 
+   * For a measured coordinate sequence a non-zero value is returned.
+   * <ul>
+   * <li>For XY sequence measures is zero</li>
+   * <li>For XYM sequence measure is one<li>
+   * <li>For XYZ sequence measure is zero</li>
+   * <li>For XYZM sequence measure is one</li>
+   * <li>Values greater than one are supported</li>
+   * </ul>
+   *
+   * @return the number of measures included in dimension
+   */
+  default int getMeasures() {
```

Comment:
This might be better named getNumberOfMeasures, or getMeasureCount

Reply:
I settled on shorter, after trying getNumberOfMeasures() initially.

Reply:
Yeah, I'll say that I'd expect getMeasures to return an Array/List of the measures rather than a count.  Can we change the name?